name: Running Testkube Tests.
on:
  pull_request:
    types:
      - opened
env:
  CLUSTER_NAME: scrumptious-badger-1698309230
  AWS_REGION: us-west-2
  
jobs:
   job_1:
     name: test workflow for AWS
     runs-on: ubuntu-latest
     outputs:
       outpu1: ${{ steps.get_output.outputs.status }}
     steps:
     - name: Configure AWS credentials
       uses: aws-actions/configure-aws-credentials@v4
       with:
         aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
         aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
         aws-region: us-west-2
     - name: kubeconfig
       run: |
         aws eks update-kubeconfig --name ${{ env.CLUSTER_NAME }} --region ${{ env.AWS_REGION }}
     - name: Create & Run K6 Test To Test Amazon.com
       uses: kubeshop/setup-testkube@v1
       env:
         KUBECONFIG: ${{ github.workspace }}/config
     - run: |
        testkube create test --name k6-amazon-test --type k6/script --test-content-type git-file --git-uri https://github.com/techmaharaj/testkube-GH.git --git-branch main --git-path k6/amazon.js
        testkube run test k6-amazon-test -f
     - name: Get Output
       id: get_output
       run: |
         exitcode="$?"
         echo "exitcode=$exitcode" >> $GITHUB_OUTPUT
         exit "$exitcode"
     - name: Check Exit Code and Act
       id: check_and_act
       run: |
          if [[ "${{ steps.get_output.outputs.exitcode }}" == "0" ]]; then
            echo "The previous step passed with exit code 0."
            echo "Closing the PR since the exit code is 0."
            curl -L \
            -X PUT \
            -H "Accept: application/vnd.github+json" \
            -H "Authorization: Bearer ${{ secrets.GH_TOKEN }}" \
            -H "X-GitHub-Api-Version: 2022-11-28" \
            https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/merge \
            -d '{"commit_title":"All Checks PAssed","commit_message":"PR auto-closed due to successful checks"}'
          else
            echo "The previous step failed with exit code ${{ steps.get_output.outputs.exitcode }}."
            exit 1
          fi

          
      
  

